Makefile

```
@@ -148,6 +148,14 @@ distclean:
 install: checkprefix nvim
 	+$(BUILD_TOOL) -C build install
 
+clang-tidy: build/.ran-cmake
+	@test $(FILE) \
+		&& clang-tidy --config-file .clang-tidy -p build $(FILE) \
+		|| $(MAKE) clang-tidy-full
```

What does `test $(FILE)` even do? It gives the same exit code regardless if the file exists or not. Is there a `-f` missing or something?

Also, I think doing 

```suggestion
	clang-tidy --config-file .clang-tidy -p build $(FILE) || $(MAKE) clang-tidy-full
```

would achieve the same thing.

---

@dundargoc, did you try it this way?

```sh
make clang-tidy FILE=src/nvim/screen.c
```

---

Yes?

---

it's good now either way, since [target has been simplified](https://github.com/neovim/neovim/pull/15601/commits/48609e73aacd1c6a104cf97a3836a2561cd8bbe8)